### PR TITLE
handle upstream annotations that contain a datacenter segment

### PR DESCRIPTION
**Pull request by DerekStrickland**

### CODE REVIEW

**Comment by DerekStrickland** [comment]
Should handle [this](https://github.com/hashicorp/consul-k8s/issues/246) issue

**Comment by shellfu** [comment]
Any status on getting this merged in at all?

File: connect-inject/container_env_test.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+			},
+		}
+	}
+
+	cases := []struct {
+		Name        string
+		Pod         func(*corev1.Pod) *corev1.Pod
+		ServiceHost string // Service host name used to craft env var names
+		ServicePort string // Service port that should be set to env var
+	}{
```
Comment (ishustava) [inline connect-inject/container_env_test.go:33]: I'd recommend just hard-coding the upstream values and also adding a testcase for an upstream without datacenter. I suggest changing this to:
```go
cases := []struct {
  name string
  upstream string
}{
  "Upstream with datacenter",
  "static-server:7890:dc1",
},
{
  "Upstream without datacenter",
  "static-server:7890",
}
```
Then you could move the minimal pod inside the `for` loop and remove the `serviceHost`, `servicePort`, and `upstreamAnnotation` variables.

```go

	for _, tt := range cases {
		t.Run(tt.Name, func(t *testing.T) {
			require := require.New(t)

			var h Handler
			envVars := h.containerEnvVars(&corev1.Pod{
			ObjectMeta: metav1.ObjectMeta{
				Annotations: map[string]string{
					annotationService: "foo",
                                        annotationUpsteams: tt.upstream,
				},
			      },
		           }
	                })

			....
}
```

File: connect-inject/container_env_test.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+					portEntryFound = true
+					if envVars[i].Value == tt.ServicePort {
+						portCorrectlySet = true
+					}
+				}
+			}
+
+			require.True(hostEntryFound)
+			require.True(portEntryFound)
+			require.True(portCorrectlySet)
```
Comment (ishustava) [inline connect-inject/container_env_test.go:75]: I think we always add two env variables per upstream. In that case, would changing this to one assertion work?

```go
require.ElementsMatch(envVars, []corev1.EnvVar{
	Name:  "STATIC_SERVER_CONNECT_SERVICE_HOST",
	Value: "127.0.0.1",
}, corev1.EnvVar{
	Name:  "STATIC_SERVER_CONNECT_SERVICE_PORT",
	Value: "7809",
})
```

File: connect-inject/container_env_test.go (hunk 1 above)
Comment (DerekStrickland) [reply connect-inject/container_env_test.go:33]: Thanks for the feedback @ishustava .  I'll take a stab in my off-hours and ping when I've updated.

Comment (DerekStrickland) [reply connect-inject/container_env_test.go]: @ishustava  I made the modifications.  Thanks for the input.  Simplified things a lot.

File: connect-inject/container_env_test.go (hunk 2 above)
Comment (DerekStrickland) [reply connect-inject/container_env_test.go]: done

**Pull request merged**